src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -444,6 +444,40 @@ public static bool TryJoin(Span<char> destination, out int charsWritten, ReadOnl
             return true;
         }
 
+        public static bool TryJoin(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2, ReadOnlySpan<char> path3, Span<char> destination, out int charsWritten)
+        {
+            charsWritten = 0;
```

Comment:
nit: Thoughts on moving this assignment only at the return sites? In this case, there are only two such places (line 451/467) since it is already set in other places.